packages/emitter-framework/src/python/components/enum-declaration/enum-declaration.test.tsx

```diff
@@ -0,0 +1,246 @@
+        two: 2,
+        three: 3
+      }
+    `);
+    const output = getOutput(runner.program, [
+      <EnumDeclaration type={result.Foo} doc={["This is an explicit doc"]} />,
+    ]);
+
+    expect(output).toRenderTo(d`
+      from enum import IntEnum
+
+      # This is an explicit doc
+      class Foo(IntEnum):
+        ONE = 1  # This is one
+        TWO = 2
+        THREE = 3
+
+
+    `);
+  });
+
+  it("takes a union type parameter", async () => {
+    const result = await runner.compile(t.code`
+      union ${t.union("Foo")} {
+        one: 1,
+        two: 2,
+        three: 3
+      }
+    `);
+    const output = getOutput(runner.program, [<EnumDeclaration type={result.Foo} />]);
+
+    expect(output).toRenderTo(d`
+      from enum import IntEnum
+
+      class Foo(IntEnum):
+        ONE = 1
+        TWO = 2
+        THREE = 3
+
+
+    `);
+  });
+
+  it("can be referenced", async () => {
+    const result = await runner.compile(t.code`
+      enum ${t.enum("Foo")} {
+        one: 1,
+        two: 2,
+        three: 3
+      }
+    `);
+
+    const output = getOutput(runner.program, [
+      <EnumDeclaration type={result.Foo} />,
+      <StatementList>
+        {efRefkey(result.Foo)}
+        {efRefkey(result.Foo.members.get("one"))}
+      </StatementList>,
+    ]);
+
+    expect(output).toRenderTo(d`
+      from enum import IntEnum
+
+      class Foo(IntEnum):
+        ONE = 1
+        TWO = 2
+        THREE = 3
+
+
+      Foo;
+      Foo.ONE;
+    `);
+  });
+
+  it("can be referenced using union", async () => {
+    const result = await runner.compile(t.code`
+      union ${t.union("Foo")}  {
+        one: 1,
+        two: 2,
+        three: 3
+      }
+    `);
+
+    const output = getOutput(runner.program, [
+      <EnumDeclaration type={result.Foo} />,
+      <StatementList>
+        {efRefkey(result.Foo)}
+        {efRefkey(result.Foo.variants.get("one"))}
+      </StatementList>,
+    ]);
+
+    expect(output).toRenderTo(d`
+      from enum import IntEnum
+
+      class Foo(IntEnum):
+        ONE = 1
+        TWO = 2
+        THREE = 3
+
+
+      Foo;
+      Foo.ONE;
+    `);
+  });
+
+  describe("Enum Type Detection", () => {
+    it("generates IntEnum for integer values", async () => {
+      const result = await runner.compile(t.code`
+        enum ${t.enum("StatusCode")} {
+          success: 200,
+          notFound: 404,
+          serverError: 500
+        }
+      `);
+      const output = getOutput(runner.program, [<EnumDeclaration type={result.StatusCode} />]);
+
+      expect(output).toRenderTo(d`
+        from enum import IntEnum
+
+        class StatusCode(IntEnum):
+          SUCCESS = 200
+          NOT_FOUND = 404
+          SERVER_ERROR = 500
+
+
+      `);
+    });
+
+    it("generates StrEnum for string values", async () => {
+      const result = await runner.compile(t.code`
+        enum ${t.enum("Color")} {
+          red: "red",
+          green: "green",
+          blue: "blue"
+        }
```

Comment:
Would also be good to have a test case where the names and values are different, to ensure that we're using the right thing.